Running an empty RawQuery causes a panic

## Description

Running an empty RawQuery causes a panic.

### Steps to Reproduce the Problem

```
sqlite, _ := pop.NewConnection(&pop.ConnectionDetails{
	URL: "sqlite3://" + filepath.Join(os.TempDir(), x.NewUUID().String()) + ".sql?mode=memory&_fk=true",
})
_ = sqlite.Open()
sqlite.RawQuery("").Exec()
```

### Expected Behavior

No panic.

### Actual Behavior

A panic happens:

```
% go test -tags sqlite .
--- FAIL: TestMigrations (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x46c74f2]

goroutine 10 [running]:
testing.tRunner.func1.1(0x4d86060, 0x5973aa0)
        /usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:940 +0x2f5
testing.tRunner.func1(0xc0001c58c0)
        /usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:943 +0x3f9
panic(0x4d86060, 0x5973aa0)
        /usr/local/Cellar/go/1.14.3/libexec/src/runtime/panic.go:969 +0x166
github.com/gobuffalo/pop/v5.(*Model).TableName(0x0, 0x0, 0x0)
        /Users/foobar/go/src/github.com/gobuffalo/pop/model.go:79 +0x52
github.com/gobuffalo/pop/v5.(*sqlBuilder).buildColumns(0xc0001fb868, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/foobar/go/src/github.com/gobuffalo/pop/sql_builder.go:218 +0x65
github.com/gobuffalo/pop/v5.(*sqlBuilder).buildSelectSQL(0xc0001fb868, 0x0, 0x0)
        /Users/foobar/go/src/github.com/gobuffalo/pop/sql_builder.go:102 +0x40
github.com/gobuffalo/pop/v5.(*sqlBuilder).compile(0xc0001fb868)
        /Users/foobar/go/src/github.com/gobuffalo/pop/sql_builder.go:87 +0x369
github.com/gobuffalo/pop/v5.(*sqlBuilder).String(...)
        /Users/foobar/go/src/github.com/gobuffalo/pop/sql_builder.go:57
github.com/gobuffalo/pop/v5.Query.ToSQL(0xc0000333b0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/foobar/go/src/github.com/gobuffalo/pop/query.go:206 +0x1db
github.com/gobuffalo/pop/v5.(*Query).Exec.func1(0xbfaffeb74d0bc090, 0xcc6278)
        /Users/foobar/go/src/github.com/gobuffalo/pop/executors.go:25 +0x91
github.com/gobuffalo/pop/v5.(*Connection).timeFunc(0xc0007a4900, 0x4f07ee0, 0x4, 0xc000079cd0, 0x30, 0x4ded6e0)
        /Users/foobar/go/src/github.com/gobuffalo/pop/connection.go:231 +0x50
github.com/gobuffalo/pop/v5.(*Query).Exec(0xc000079e70, 0xc0000333b0, 0x0)
        /Users/foobar/go/src/github.com/gobuffalo/pop/executors.go:24 +0x6a
github.com/ory/kratos/persistence/sql/migratest.TestMigrations(0xc0001c58c0)
        /Users/foobar/go/src/github.com/ory/kratos/persistence/sql/migratest/migration_test.go:74 +0x2f7
testing.tRunner(0xc0001c58c0, 0x4f7c148)
        /usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:991 +0xdc
created by testing.(*T).Run
        /usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:1042 +0x357
FAIL    github.com/ory/kratos/persistence/sql/migratest 0.429s
```

### Info

```
github.com/gobuffalo/pop/v5 v5.1.3
```

--- a/executors.go
+++ b/executors.go
@@ -20,10 +20,16 @@ func (c *Connection) Reload(model interface{}) error {
 	})
 }
 
+// TODO: consider merging the following two methods.
+
 // Exec runs the given query.
 func (q *Query) Exec() error {
 	return q.Connection.timeFunc("Exec", func() error {
 		sql, args := q.ToSQL(nil)
+		if sql == "" {
+			return fmt.Errorf("empty query")
+		}
+
 		log(logging.SQL, sql, args...)
 		_, err := q.Connection.Store.Exec(sql, args...)
 		return err
@@ -36,6 +42,10 @@ func (q *Query) ExecWithCount() (int, error) {
 	count := int64(0)
 	return int(count), q.Connection.timeFunc("Exec", func() error {
 		sql, args := q.ToSQL(nil)
+		if sql == "" {
+			return fmt.Errorf("empty query")
+		}
+
 		log(logging.SQL, sql, args...)
 		result, err := q.Connection.Store.Exec(sql, args...)
 		if err != nil {

--- a/query.go
+++ b/query.go
@@ -213,6 +213,10 @@ func Q(c *Connection) *Query {
 // from the `Model` passed in.
 func (q Query) ToSQL(model *Model, addColumns ...string) (string, []interface{}) {
 	sb := q.toSQLBuilder(model, addColumns...)
+	// nil model is allowed only when if RawSQL is provided.
+	if model == nil && (q.RawSQL == nil || q.RawSQL.Fragment == "") {
+		return "", nil
+	}
 	return sb.String(), sb.Args()
 }
 

--- a/query_test.go
+++ b/query_test.go
@@ -386,3 +386,22 @@ func Test_ToSQL_RawQuery(t *testing.T) {
 		a.Equal(args, []interface{}{"random", "query"})
 	})
 }
+
+func Test_RawQuery_Empty(t *testing.T) {
+	Debug = true
+	defer func() { Debug = false }()
+
+	t.Run("EmptyQuery", func(t *testing.T) {
+		r := require.New(t)
+		transaction(func(tx *Connection) {
+			r.Error(tx.Q().Exec())
+		})
+	})
+
+	t.Run("EmptyRawQuery", func(t *testing.T) {
+		r := require.New(t)
+		transaction(func(tx *Connection) {
+			r.Error(tx.RawQuery("").Exec())
+		})
+	})
+}

--- a/sql_builder.go
+++ b/sql_builder.go
@@ -17,6 +17,8 @@ type sqlBuilder struct {
 	AddColumns []string
 	sql        string
 	args       []interface{}
+	isCompiled bool
+	err        error
 }
 
 func newSQLBuilder(q Query, m *Model, addColumns ...string) *sqlBuilder {
@@ -25,6 +27,7 @@ func newSQLBuilder(q Query, m *Model, addColumns ...string) *sqlBuilder {
 		Model:      m,
 		AddColumns: addColumns,
 		args:       []interface{}{},
+		isCompiled: false,
 	}
 }
 
@@ -53,19 +56,15 @@ func hasLimitOrOffset(sqlString string) bool {
 }
 
 func (sq *sqlBuilder) String() string {
-	if sq.sql == "" {
+	if !sq.isCompiled {
 		sq.compile()
 	}
 	return sq.sql
 }
 
 func (sq *sqlBuilder) Args() []interface{} {
-	if len(sq.args) == 0 {
-		if len(sq.Query.RawSQL.Arguments) > 0 {
-			sq.args = sq.Query.RawSQL.Arguments
-		} else {
-			sq.compile()
-		}
+	if !sq.isCompiled {
+		sq.compile()
 	}
 	return sq.args
 }
@@ -83,7 +82,12 @@ func (sq *sqlBuilder) compile() {
 				}
 				sq.sql = sq.Query.RawSQL.Fragment
 			}
+			sq.args = sq.Query.RawSQL.Arguments
 		} else {
+			if sq.Model == nil {
+				sq.err = fmt.Errorf("sqlBuilder.compile() called but no RawSQL and Model specified")
+				return
+			}
 			switch sq.Query.Operation {
 			case Select:
 				sq.sql = sq.buildSelectSQL()

